docs/articles_en/documentation/openvino-extensibility/transformation-api/patterns-python-api.rst

```diff
@@ -0,0 +1,130 @@
+
+Our code already looks promissing, however in OpenVINO™ we usually don't create patterns using the same nodes we used for creating the model. Instead, we want to use so-called wrappers that provide us for additional functionality.
+For the given case we would probably use ``WrapType`` and the code would look as following:
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: cpp
+   :fragment: [ov:create_simple_model_and_pattern_wrap_type]
+
+1. WrapType
+++++++++++++++++++++++++++++++++++++++++
+
+``WrapType`` is a wrapper used to store one or many types to match them. As we already saw, it is possible to specify a single type in ``WrapType`` and use it for matching.
+However, it is also possible to list all possible types for the given node. For example, you may do something like this:
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: cpp
+   :fragment: [ov:wrap_type_list]
+
+As you may see, ``pattern_sig`` is created with the list ``["opset13.Relu", "opset13.Sigmoid"]`` which means it can either be a ``Relu`` or ``Sigmoid``. Pretty convenient, huh?
+This is why matching the same pattern against different nodes becomes possible. Basically, we may think of ``WrapType`` as "one of listed". Note, that you may provide more than 2 types
+to ``WrapType``.
+
+If you want to have some additional checking for you node, you may create a predicate for it providing a function or a lambda. This function will be executed during
+matching performing the additional validation specified in the logic of the function. For example, you may want to check the consumers count of a given node:
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: cpp
+   :fragment: [ov:wrap_type_predicate]
+
+2. AnyInput 
+++++++++++++++++++++++++++++++++++++++++
+You have already seen ``AnyInput`` in the above examples. We use it when we don't really care about a specific input for a given node.
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: cpp
+   :fragment: [ov:any_input]
+
+You may also create ``AnyInput()`` with a predicate, if you want some additional checks for you input. It would look similar to ``WrapType`` with a lambda or a function. Let's say we want to make sure the inputs has a rank of 4.
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: cpp
+   :fragment: [ov:any_input_predicate]
+
+3. Or
+++++++++++++++++++++++++++++++++++++++++
+``Or`` is somewhat similar to ``WrapType``, however if ``WrapType`` can only match one of types provided in the list, ``Or`` is used to match different _branches_ of nodes.
+It would be much easier to understand with a visualization. Let's say, we want to try to match the model against two different sequences of nodes. The ``Or`` type
+facilitates this by creating 2 different branches. It would look as following:
```

Comment:
As I remember, we are supporting more than 2 branches in Or? 
I believe it's worth to mention it.

Reply:
Fixed.